src/adapters/bidderFactory.js

```diff
@@ -370,13 +368,19 @@ export function isValid(adUnitCode, bid, bidRequests) {
     return `Invalid bid from ${bid.bidderCode}. Ignoring bid: ${msg}`;
   }
 
+  if (!bid) {
+    logError(`Some adapter tried to add an undefined bid for ${adUnitCode}.`);
+    return false;
+  }
+
   if (!adUnitCode) {
-    logWarn('No adUnitCode was supplied to addBidResponse.');
+    logError(errorMessage('No adUnitCode was supplied to addBidResponse.'));
     return false;
   }
 
-  if (!bid) {
-    logWarn(`Some adapter tried to add an undefined bid for ${adUnitCode}.`);
+  const req = getBidderRequest(bidRequests, bid.bidderCode, adUnitCode);
```

Comment:
@kitwestneat We have done this check here https://github.com/prebid/Prebid.js/blob/master/src/adapters/bidderFactory.js#L293. If matching bid is not found than it won't reach here. Any specific use case you have in mind to do extra check here.

Reply:
Ah in the pre-1.0 version the check was in isValid, so I missed that it was moved to addBidUsingRequestMap. 

Is there a reason it was moved from isValid? It seems like the validation checks should be located together, no?

Reply:
I think the motivation was to return as early as possible when you find mismatch. isValid is used now to validate bid object key value pairs for different formats.

Reply:
ah I see. What doesn't make sense to me is that from the comments in bidderFactory, all the bidderFactory code is supposed to be semi-temporary glue. But the way it is written now, addBidResponse has this implicit dependency on the validation code in bidderFactory -- they are more tightly coupled than the comments imply. It seems to me that the bid validation should be done in addBidResponse since it is ultimately responsible for maintaining the integrity of the auction. 

It looks like the other user of addBidResponse is the s2s adapters. Why do they do their own validation? Why don't they use the same validation as bidderFactory adapters? Certainly there are validation checks that must be done on both, no?

It's not really a big deal, just confusing to me. I'll close this PR since it's not going to be useful.